onnxruntime/core/platform/path_lib.cc

```diff
@@ -16,11 +16,11 @@
 #if _GAMING_XBOX
 // Hacky, but the PathCch* APIs work on Xbox. Presumably PathCch.h needs to be updated to include the 
 // GAMES partition. It would be worthwhile to investigate this a bit more (or just use std::filesystem).
+#pragma push_macro("WINAPI_FAMILY")
```

Comment:
For the code that solely written for the Windows platform, please feel free to use std::filesystem.

Reply:
On the other platforms we have incomplete C++17 support.

Reply:
There's a small risk involved in switch the WIN32 path to std::filesystem: the PathCch* APIs handle UNC paths and forward slashes a little differently, so the std::filesystem logic would need to be carefully compliant with the existing behavior. I'd feel more comfortable punting this unless there's a good reason to disturb the existing code. This slightly hack macro push/pop won't affect non-GDK flavors so it carries zero risk.